src/data_access/api/SpotifyAPI.java

```diff
@@ -0,0 +1,109 @@
+package data_access.api;
+
+import data_access.FileAPIDataAccessObject;
+import okhttp3.*;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Random;
+
+/**
+ Example usage:
+ SpotifyAPI sapi = new SpotifyAPI();
+ ArrayList<HashMap> songs = sapi.getSongs();
+ for(HashMap s: songs){
+    System.out.println(s);
+ }
+ **/
+public class SpotifyAPI{
+    private final FileAPIDataAccessObject APIDATA;
+    private final String GET_SEARCH_BASE_ENDPOINT;
+    private final String POST_AUTH_TOKEN_ENDPOINT;
+
+
+    public SpotifyAPI(){
+        this.GET_SEARCH_BASE_ENDPOINT = "https://api.spotify.com/v1/search?q=";
+        this.POST_AUTH_TOKEN_ENDPOINT = "https://accounts.spotify.com/api/token";
+        try {
+            this.APIDATA = new FileAPIDataAccessObject("./spotify_api.csv");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public ArrayList<HashMap> getSongs(String genre){
+        return getSongsCall(genre);
+    }
+    public ArrayList<HashMap> getSongs(){
+        return getSongsCall("hiphop");
+    }
+    private ArrayList<HashMap> getSongsCall(String genre){
```

Comment:
100% agree, I didn't implement that here since I didn't want to make the Song entity and take away from someone elses work. Once we create the enetities we will adjust this for sure.